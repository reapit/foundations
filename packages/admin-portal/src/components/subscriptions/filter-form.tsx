import React, { Dispatch, FC, SetStateAction } from 'react'
import { useForm } from 'react-hook-form'
import { FormLayout, InputWrap, InputGroup, Label, Select } from '@reapit/elements'
import { SubscriptionsFilters } from '.'
import debounce from 'just-debounce-it'

export interface FilterFormProps {
  setSubscriptionsFilters: Dispatch<SetStateAction<SubscriptionsFilters>>
}

export const FilterForm: FC<FilterFormProps> = ({ setSubscriptionsFilters }) => {
  const { register, handleSubmit } = useForm<SubscriptionsFilters>({ mode: 'all' })

  return (
    <form onChange={handleSubmit(debounce(setSubscriptionsFilters, 500))}>
      <FormLayout hasMargin>
        <InputWrap>
          <InputGroup {...register('organisationName')} label="Company Name" />
        </InputWrap>
        <InputWrap>
          <InputGroup {...register('userEmail')} type="email" label="User Email" />
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Label>Subscription Type</Label>
            <Select {...register('subscriptionType')}>
              <option value="">Please Select</option>
              <option value="applicationListing">Application Listing</option>
              <option value="developerRegistration">Developer Registration</option>
              <option value="developerEdition">Developer Edition</option>
            </Select>
          </InputGroup>
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Label>Status</Label>
            <Select {...register('status')}>
              <option value="">Please Select</option>
              <option value="active">Active</option>
              <option value="cancelled">Cancelled</option>
            </Select>
          </InputGroup>
        </InputWrap>
      </FormLayout>
    </form>
  )
}
