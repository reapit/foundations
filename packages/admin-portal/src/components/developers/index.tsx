import React, { Dispatch, FC, SetStateAction, useState } from 'react'
import ErrorBoundary from '../error-boundary'
import { DeveloperFilters, FilterForm } from './filter-form'
import { DeveloperModelPagedResult } from '@reapit/foundations-ts-definitions'
import { reapitConnectBrowserSession } from '../../core/connect-session'
import { Loader, PageContainer, Pagination, Title } from '@reapit/elements'
import { Statistics } from '../statistics'
import { objectToQuery, useReapitGet, GetActionNames, getActions } from '@reapit/use-reapit-data'
import { DevelopersTable } from './developers-table'

export const closeDisableMemberModal = (setDisableMemberModalVisible: Dispatch<SetStateAction<boolean>>) => () => {
  setDisableMemberModalVisible(false)
}

export const DevsManagement: FC = () => {
  const [developerFilters, setDeveloperFilters] = useState<DeveloperFilters>({})
  const [pageNumber, setPageNumber] = useState<number>(1)
  const [pageSize, setPageSize] = useState<number>(12)

  const [developers, developersLoading, , refreshDevelopers] = useReapitGet<DeveloperModelPagedResult>({
    reapitConnectBrowserSession,
    action: getActions[GetActionNames.getDevelopers],
    queryParams: { pageNumber, pageSize, ...objectToQuery(developerFilters) },
  })

  return (
    <ErrorBoundary>
      <PageContainer>
        <Title>Developers</Title>
        <FilterForm setDeveloperFilters={setDeveloperFilters} />
        {developersLoading ? (
          <Loader />
        ) : (
          <>
            <Statistics area="DEVELOPERS" data={developers} setPageSize={setPageSize} />
            <DevelopersTable developers={developers} refreshDevelopers={refreshDevelopers} />
            <Pagination
              callback={setPageNumber}
              currentPage={pageNumber}
              numberPages={Math.ceil((developers?.totalCount ?? 1) / 12)}
            />
          </>
        )}
      </PageContainer>
    </ErrorBoundary>
  )
}

export default DevsManagement
