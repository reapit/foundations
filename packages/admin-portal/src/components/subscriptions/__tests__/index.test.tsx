import React from 'react'
import Subscriptions, { handleCancelSub, handleCancelSubSuccess, handleSetSubId } from '../index'
import { render } from '../../../tests/react-testing'
import { mockDeveloperModelPagedResult } from '../../../tests/__stubs__/developers'
import { mockSubscriptionModelPagedResult } from '../../../tests/__stubs__/subscriptions'
import { useReapitGet } from '@reapit/utils-react'

jest.mock('../../../core/use-permissions-state')
jest.mock('@reapit/utils-react', () => ({
  useReapitGet: jest.fn(() => [null, false]),
  useReapitUpdate: jest.fn(() => [undefined, undefined, jest.fn()]),
  objectToQuery: jest.fn(),
}))

const mockUseReapitGet = useReapitGet as jest.Mock

describe('Subscriptions', () => {
  it('should render component when loading', () => {
    mockUseReapitGet.mockReturnValue([null, true])
    expect(render(<Subscriptions />)).toMatchSnapshot()
  })

  it('should render component when has data', () => {
    mockUseReapitGet
      .mockReturnValue([mockSubscriptionModelPagedResult, false])
      .mockReturnValue([mockDeveloperModelPagedResult, false])
    expect(render(<Subscriptions />)).toMatchSnapshot()
  })
})

describe('handleCancelSub', () => {
  it('handleCancelSub should  correctly set the cancel sub id', () => {
    const cancelSub = jest.fn()
    const setCancelSubId = jest.fn()
    const curried = handleCancelSub(cancelSub, setCancelSubId)

    curried()

    expect(cancelSub).toHaveBeenCalledTimes(1)
    expect(setCancelSubId).toHaveBeenCalledWith(null)
  })
})
describe('handleCancelSubSuccess', () => {
  it('handleCancelSubSuccess should refresh subs on success', () => {
    const refetchSubs = jest.fn()
    const closeModal = jest.fn()
    const success = true
    const curried = handleCancelSubSuccess(refetchSubs, closeModal, success)

    curried()

    expect(refetchSubs).toHaveBeenCalledTimes(1)
    expect(closeModal).toHaveBeenCalledTimes(1)
  })
})

describe('handleSetSubId', () => {
  it('handleSetSubId should correctly set the sub id', () => {
    const setCancelSubId = jest.fn()
    const openModal = jest.fn()
    const cancelSubId = 'MOCK_ID'
    const curried = handleSetSubId(setCancelSubId, openModal, cancelSubId)

    curried()

    expect(openModal).toHaveBeenCalledTimes(1)
    expect(setCancelSubId).toHaveBeenCalledWith(cancelSubId)
  })
})
