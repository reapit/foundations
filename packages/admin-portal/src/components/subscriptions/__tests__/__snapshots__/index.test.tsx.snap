// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscriptions should render component when has data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-93 el-snack-holder"
      />
      <section
        class="el-page-container"
      >
        <h1
          class="el-text-base el-title"
        >
          Subscriptions
        </h1>
        <form>
          <div
            class="el-form-layout-has-margin el-form-layout"
          >
            <div
              class="mocked-styled-133 el-input-wrap"
            >
              <div
                class="mocked-styled-40 el-input-group"
              >
                <input
                  aria-label="Input type undefined"
                  class="mocked-styled-38 el-input"
                  id="test-static-id"
                  name="organisationName"
                />
                <label
                  class="mocked-styled-41 el-label"
                  for="test-static-id"
                >
                  Company Name
                </label>
              </div>
            </div>
            <div
              class="mocked-styled-133 el-input-wrap"
            >
              <div
                class="mocked-styled-40 el-input-group"
              >
                <input
                  aria-label="Input type email"
                  class="mocked-styled-38 el-input"
                  id="test-static-id"
                  name="userEmail"
                  type="email"
                />
                <label
                  class="mocked-styled-41 el-label"
                  for="test-static-id"
                >
                  User Email
                </label>
              </div>
            </div>
            <div
              class="mocked-styled-133 el-input-wrap"
            >
              <div
                class="mocked-styled-40 el-input-group"
              >
                <label
                  class="mocked-styled-41 el-label"
                >
                  Subscription Type
                </label>
                <select
                  aria-label="Select an item from the dropdown list"
                  class="mocked-styled-83 el-select"
                  name="subscriptionType"
                >
                  <option
                    value=""
                  >
                    Please Select
                  </option>
                  <option
                    value="applicationListing"
                  >
                    Application Listing
                  </option>
                  <option
                    value="developerRegistration"
                  >
                    Developer Registration
                  </option>
                  <option
                    value="developerEdition"
                  >
                    Developer Edition
                  </option>
                </select>
              </div>
            </div>
            <div
              class="mocked-styled-133 el-input-wrap"
            >
              <div
                class="mocked-styled-40 el-input-group"
              >
                <label
                  class="mocked-styled-41 el-label"
                >
                  Status
                </label>
                <select
                  aria-label="Select an item from the dropdown list"
                  class="mocked-styled-83 el-select"
                  name="status"
                >
                  <option
                    value=""
                  >
                    Please Select
                  </option>
                  <option
                    value="active"
                  >
                    Active
                  </option>
                  <option
                    value="cancelled"
                  >
                    Cancelled
                  </option>
                </select>
              </div>
            </div>
          </div>
        </form>
        <div
          class="el-mb11"
        >
          <div
            class="mocked-styled-2 el-button-group"
          >
            <div
              class="mocked-styled-3 el-button-group-inner"
            >
              <button
                class="el-intent-primary el-button"
                type="button"
              >
                <div
                  class="mocked-styled-0 el-button-loader"
                />
                Download CSV
              </button>
            </div>
          </div>
        </div>
        <div
          class="el-mb11 el-table"
          data-has-call-to-action="true"
          data-has-expandable-action="false"
          data-num-columns-excl-action-col="9"
          role="table"
        >
          <div
            class="mocked-styled-101 el-table-headers-row"
            role="row"
          >
            <div
              aria-label="Subcription Type"
              class="mocked-styled-102 el-table-header"
              role="columnheader"
            >
              Subcription Type
            </div>
            <div
              aria-label="Summary"
              class="mocked-styled-102 el-table-header"
              role="columnheader"
            >
              Summary
            </div>
            <div
              aria-label="Company Name"
              class="mocked-styled-102 el-table-header"
              role="columnheader"
            >
              Company Name
            </div>
            <div
              aria-label="User Email"
              class="mocked-styled-102 el-table-header"
              role="columnheader"
            >
              User Email
            </div>
            <div
              aria-label="Start Date"
              class="mocked-styled-102 el-table-header"
              role="columnheader"
            >
              Start Date
            </div>
            <div
              aria-label="Renews"
              class="mocked-styled-102 el-table-header"
              role="columnheader"
            >
              Renews
            </div>
            <div
              aria-label="Frequency"
              class="mocked-styled-102 el-table-header"
              role="columnheader"
            >
              Frequency
            </div>
            <div
              aria-label="Cost"
              class="mocked-styled-102 el-table-header"
              role="columnheader"
            >
              Cost
            </div>
            <div
              aria-label="Cancelled"
              class="mocked-styled-102 el-table-header"
              role="columnheader"
            >
              Cancelled
            </div>
            <div
              class="mocked-styled-102 el-table-header"
              role="columnheader"
            >
              Cancel
            </div>
          </div>
          <div
            class="mocked-styled-110 el-table-row-container"
            role="row"
          >
            <div
              class="mocked-styled-103 el-table-row"
            >
              <div
                class="el-table-cell-has-dark-text el-table-cell"
                role="cell"
              >
                <div
                  class="mocked-styled-111 el-table-cell-content"
                  data-narrow-label="Subcription Type"
                />
              </div>
              <div
                class="el-table-cell-has-dark-text el-table-cell"
                role="cell"
              >
                <div
                  class="mocked-styled-111 el-table-cell-content"
                  data-narrow-label="Summary"
                />
              </div>
              <div
                class="el-table-cell-has-dark-text el-table-cell"
                role="cell"
              >
                <div
                  class="mocked-styled-111 el-table-cell-content"
                  data-narrow-label="Company Name"
                />
              </div>
              <div
                class="el-table-cell-has-dark-text el-table-cell"
                role="cell"
              >
                <div
                  class="mocked-styled-111 el-table-cell-content"
                  data-narrow-label="User Email"
                />
              </div>
              <div
                class="el-table-cell-has-dark-text el-table-cell"
                role="cell"
              >
                <div
                  class="mocked-styled-111 el-table-cell-content"
                  data-narrow-label="Start Date"
                >
                  03 Mar 2022 10:18
                </div>
              </div>
              <div
                class="el-table-cell-has-dark-text el-table-cell"
                role="cell"
              >
                <div
                  class="mocked-styled-111 el-table-cell-content"
                  data-narrow-label="Renews"
                >
                  -
                </div>
              </div>
              <div
                class="el-table-cell-has-dark-text el-table-cell"
                role="cell"
              >
                <div
                  class="mocked-styled-111 el-table-cell-content"
                  data-narrow-label="Frequency"
                />
              </div>
              <div
                class="el-table-cell-has-dark-text el-table-cell"
                role="cell"
              >
                <div
                  class="mocked-styled-111 el-table-cell-content"
                  data-narrow-label="Cost"
                >
                  Â£undefined
                </div>
              </div>
              <div
                class="el-table-cell-has-dark-text el-table-cell"
                role="cell"
              >
                <div
                  class="mocked-styled-111 el-table-cell-content"
                  data-narrow-label="Cancelled"
                >
                  -
                </div>
              </div>
              <div
                class="mocked-styled-104 el-table-cta-cell"
                role="cell"
                tabindex="0"
              />
            </div>
          </div>
        </div>
        <div
          class="mocked-styled-71 el-pagination-wrap"
        >
          <button
            aria-label="Previous"
            class="mocked-styled-74 el-pagination-button"
            data-testid="back-button"
            role="button"
            type="button"
          >
            <span
              class="el-icon-new el-icon"
            >
              <svg
                fill="none"
                height="1em"
                role="img"
                title="Icon image with name chevronLeft"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.496 20.004a1 1 0 0 1-.71-.29l-6.99-7a1 1 0 0 1 0-1.42l6.99-7a1.002 1.002 0 0 1 1.678.45c.093.347-.006.716-.26.97l-6.291 6.29 6.292 6.29a1 1 0 0 1-.71 1.71"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
          <div
            class="mocked-styled-72 el-pagination-text"
          >
            <input
              aria-label="Current page number, edit to select page of 1 available"
              class="mocked-styled-73 el-pagination-input"
              value="1"
            />
             
            of 
            1
          </div>
          <button
            aria-label="Next"
            class="mocked-styled-74 el-pagination-button"
            data-testid="forward-button"
            role="button"
            type="button"
          >
            <span
              class="el-icon-new el-icon"
            >
              <svg
                fill="none"
                height="1em"
                role="img"
                title="Icon image with name chevronRight"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.505 20.004a1 1 0 0 1-1.005-1 1 1 0 0 1 .295-.71l6.292-6.29-6.292-6.29a1.005 1.005 0 0 1 0-1.42 1 1 0 0 1 1.419 0l6.99 7a1 1 0 0 1 0 1.42l-6.99 7a1 1 0 0 1-.71.29"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-93 el-snack-holder"
    />
    <section
      class="el-page-container"
    >
      <h1
        class="el-text-base el-title"
      >
        Subscriptions
      </h1>
      <form>
        <div
          class="el-form-layout-has-margin el-form-layout"
        >
          <div
            class="mocked-styled-133 el-input-wrap"
          >
            <div
              class="mocked-styled-40 el-input-group"
            >
              <input
                aria-label="Input type undefined"
                class="mocked-styled-38 el-input"
                id="test-static-id"
                name="organisationName"
              />
              <label
                class="mocked-styled-41 el-label"
                for="test-static-id"
              >
                Company Name
              </label>
            </div>
          </div>
          <div
            class="mocked-styled-133 el-input-wrap"
          >
            <div
              class="mocked-styled-40 el-input-group"
            >
              <input
                aria-label="Input type email"
                class="mocked-styled-38 el-input"
                id="test-static-id"
                name="userEmail"
                type="email"
              />
              <label
                class="mocked-styled-41 el-label"
                for="test-static-id"
              >
                User Email
              </label>
            </div>
          </div>
          <div
            class="mocked-styled-133 el-input-wrap"
          >
            <div
              class="mocked-styled-40 el-input-group"
            >
              <label
                class="mocked-styled-41 el-label"
              >
                Subscription Type
              </label>
              <select
                aria-label="Select an item from the dropdown list"
                class="mocked-styled-83 el-select"
                name="subscriptionType"
              >
                <option
                  value=""
                >
                  Please Select
                </option>
                <option
                  value="applicationListing"
                >
                  Application Listing
                </option>
                <option
                  value="developerRegistration"
                >
                  Developer Registration
                </option>
                <option
                  value="developerEdition"
                >
                  Developer Edition
                </option>
              </select>
            </div>
          </div>
          <div
            class="mocked-styled-133 el-input-wrap"
          >
            <div
              class="mocked-styled-40 el-input-group"
            >
              <label
                class="mocked-styled-41 el-label"
              >
                Status
              </label>
              <select
                aria-label="Select an item from the dropdown list"
                class="mocked-styled-83 el-select"
                name="status"
              >
                <option
                  value=""
                >
                  Please Select
                </option>
                <option
                  value="active"
                >
                  Active
                </option>
                <option
                  value="cancelled"
                >
                  Cancelled
                </option>
              </select>
            </div>
          </div>
        </div>
      </form>
      <div
        class="el-mb11"
      >
        <div
          class="mocked-styled-2 el-button-group"
        >
          <div
            class="mocked-styled-3 el-button-group-inner"
          >
            <button
              class="el-intent-primary el-button"
              type="button"
            >
              <div
                class="mocked-styled-0 el-button-loader"
              />
              Download CSV
            </button>
          </div>
        </div>
      </div>
      <div
        class="el-mb11 el-table"
        data-has-call-to-action="true"
        data-has-expandable-action="false"
        data-num-columns-excl-action-col="9"
        role="table"
      >
        <div
          class="mocked-styled-101 el-table-headers-row"
          role="row"
        >
          <div
            aria-label="Subcription Type"
            class="mocked-styled-102 el-table-header"
            role="columnheader"
          >
            Subcription Type
          </div>
          <div
            aria-label="Summary"
            class="mocked-styled-102 el-table-header"
            role="columnheader"
          >
            Summary
          </div>
          <div
            aria-label="Company Name"
            class="mocked-styled-102 el-table-header"
            role="columnheader"
          >
            Company Name
          </div>
          <div
            aria-label="User Email"
            class="mocked-styled-102 el-table-header"
            role="columnheader"
          >
            User Email
          </div>
          <div
            aria-label="Start Date"
            class="mocked-styled-102 el-table-header"
            role="columnheader"
          >
            Start Date
          </div>
          <div
            aria-label="Renews"
            class="mocked-styled-102 el-table-header"
            role="columnheader"
          >
            Renews
          </div>
          <div
            aria-label="Frequency"
            class="mocked-styled-102 el-table-header"
            role="columnheader"
          >
            Frequency
          </div>
          <div
            aria-label="Cost"
            class="mocked-styled-102 el-table-header"
            role="columnheader"
          >
            Cost
          </div>
          <div
            aria-label="Cancelled"
            class="mocked-styled-102 el-table-header"
            role="columnheader"
          >
            Cancelled
          </div>
          <div
            class="mocked-styled-102 el-table-header"
            role="columnheader"
          >
            Cancel
          </div>
        </div>
        <div
          class="mocked-styled-110 el-table-row-container"
          role="row"
        >
          <div
            class="mocked-styled-103 el-table-row"
          >
            <div
              class="el-table-cell-has-dark-text el-table-cell"
              role="cell"
            >
              <div
                class="mocked-styled-111 el-table-cell-content"
                data-narrow-label="Subcription Type"
              />
            </div>
            <div
              class="el-table-cell-has-dark-text el-table-cell"
              role="cell"
            >
              <div
                class="mocked-styled-111 el-table-cell-content"
                data-narrow-label="Summary"
              />
            </div>
            <div
              class="el-table-cell-has-dark-text el-table-cell"
              role="cell"
            >
              <div
                class="mocked-styled-111 el-table-cell-content"
                data-narrow-label="Company Name"
              />
            </div>
            <div
              class="el-table-cell-has-dark-text el-table-cell"
              role="cell"
            >
              <div
                class="mocked-styled-111 el-table-cell-content"
                data-narrow-label="User Email"
              />
            </div>
            <div
              class="el-table-cell-has-dark-text el-table-cell"
              role="cell"
            >
              <div
                class="mocked-styled-111 el-table-cell-content"
                data-narrow-label="Start Date"
              >
                03 Mar 2022 10:18
              </div>
            </div>
            <div
              class="el-table-cell-has-dark-text el-table-cell"
              role="cell"
            >
              <div
                class="mocked-styled-111 el-table-cell-content"
                data-narrow-label="Renews"
              >
                -
              </div>
            </div>
            <div
              class="el-table-cell-has-dark-text el-table-cell"
              role="cell"
            >
              <div
                class="mocked-styled-111 el-table-cell-content"
                data-narrow-label="Frequency"
              />
            </div>
            <div
              class="el-table-cell-has-dark-text el-table-cell"
              role="cell"
            >
              <div
                class="mocked-styled-111 el-table-cell-content"
                data-narrow-label="Cost"
              >
                Â£undefined
              </div>
            </div>
            <div
              class="el-table-cell-has-dark-text el-table-cell"
              role="cell"
            >
              <div
                class="mocked-styled-111 el-table-cell-content"
                data-narrow-label="Cancelled"
              >
                -
              </div>
            </div>
            <div
              class="mocked-styled-104 el-table-cta-cell"
              role="cell"
              tabindex="0"
            />
          </div>
        </div>
      </div>
      <div
        class="mocked-styled-71 el-pagination-wrap"
      >
        <button
          aria-label="Previous"
          class="mocked-styled-74 el-pagination-button"
          data-testid="back-button"
          role="button"
          type="button"
        >
          <span
            class="el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name chevronLeft"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.496 20.004a1 1 0 0 1-.71-.29l-6.99-7a1 1 0 0 1 0-1.42l6.99-7a1.002 1.002 0 0 1 1.678.45c.093.347-.006.716-.26.97l-6.291 6.29 6.292 6.29a1 1 0 0 1-.71 1.71"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
        <div
          class="mocked-styled-72 el-pagination-text"
        >
          <input
            aria-label="Current page number, edit to select page of 1 available"
            class="mocked-styled-73 el-pagination-input"
            value="1"
          />
           
          of 
          1
        </div>
        <button
          aria-label="Next"
          class="mocked-styled-74 el-pagination-button"
          data-testid="forward-button"
          role="button"
          type="button"
        >
          <span
            class="el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name chevronRight"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.505 20.004a1 1 0 0 1-1.005-1 1 1 0 0 1 .295-.71l6.292-6.29-6.292-6.29a1.005 1.005 0 0 1 0-1.42 1 1 0 0 1 1.419 0l6.99 7a1 1 0 0 1 0 1.42l-6.99 7a1 1 0 0 1-.71.29"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Subscriptions should render component when loading 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-93 el-snack-holder"
      />
      <section
        class="el-page-container"
      >
        <h1
          class="el-text-base el-title"
        >
          Subscriptions
        </h1>
        <form>
          <div
            class="el-form-layout-has-margin el-form-layout"
          >
            <div
              class="mocked-styled-133 el-input-wrap"
            >
              <div
                class="mocked-styled-40 el-input-group"
              >
                <input
                  aria-label="Input type undefined"
                  class="mocked-styled-38 el-input"
                  id="test-static-id"
                  name="organisationName"
                />
                <label
                  class="mocked-styled-41 el-label"
                  for="test-static-id"
                >
                  Company Name
                </label>
              </div>
            </div>
            <div
              class="mocked-styled-133 el-input-wrap"
            >
              <div
                class="mocked-styled-40 el-input-group"
              >
                <input
                  aria-label="Input type email"
                  class="mocked-styled-38 el-input"
                  id="test-static-id"
                  name="userEmail"
                  type="email"
                />
                <label
                  class="mocked-styled-41 el-label"
                  for="test-static-id"
                >
                  User Email
                </label>
              </div>
            </div>
            <div
              class="mocked-styled-133 el-input-wrap"
            >
              <div
                class="mocked-styled-40 el-input-group"
              >
                <label
                  class="mocked-styled-41 el-label"
                >
                  Subscription Type
                </label>
                <select
                  aria-label="Select an item from the dropdown list"
                  class="mocked-styled-83 el-select"
                  name="subscriptionType"
                >
                  <option
                    value=""
                  >
                    Please Select
                  </option>
                  <option
                    value="applicationListing"
                  >
                    Application Listing
                  </option>
                  <option
                    value="developerRegistration"
                  >
                    Developer Registration
                  </option>
                  <option
                    value="developerEdition"
                  >
                    Developer Edition
                  </option>
                </select>
              </div>
            </div>
            <div
              class="mocked-styled-133 el-input-wrap"
            >
              <div
                class="mocked-styled-40 el-input-group"
              >
                <label
                  class="mocked-styled-41 el-label"
                >
                  Status
                </label>
                <select
                  aria-label="Select an item from the dropdown list"
                  class="mocked-styled-83 el-select"
                  name="status"
                >
                  <option
                    value=""
                  >
                    Please Select
                  </option>
                  <option
                    value="active"
                  >
                    Active
                  </option>
                  <option
                    value="cancelled"
                  >
                    Cancelled
                  </option>
                </select>
              </div>
            </div>
          </div>
        </form>
        <div
          class="el-mb11"
        >
          <div
            class="mocked-styled-2 el-button-group"
          >
            <div
              class="mocked-styled-3 el-button-group-inner"
            >
              <button
                class="el-intent-primary el-button"
                type="button"
              >
                <div
                  class="mocked-styled-0 el-button-loader"
                />
                Download CSV
              </button>
            </div>
          </div>
        </div>
        <div
          class="mocked-styled-44 el-loader-container"
        >
          <div
            class="mocked-styled-46 el-loader"
          >
            <div
              class="mocked-styled-47 el-loader-moving-bar"
            />
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-93 el-snack-holder"
    />
    <section
      class="el-page-container"
    >
      <h1
        class="el-text-base el-title"
      >
        Subscriptions
      </h1>
      <form>
        <div
          class="el-form-layout-has-margin el-form-layout"
        >
          <div
            class="mocked-styled-133 el-input-wrap"
          >
            <div
              class="mocked-styled-40 el-input-group"
            >
              <input
                aria-label="Input type undefined"
                class="mocked-styled-38 el-input"
                id="test-static-id"
                name="organisationName"
              />
              <label
                class="mocked-styled-41 el-label"
                for="test-static-id"
              >
                Company Name
              </label>
            </div>
          </div>
          <div
            class="mocked-styled-133 el-input-wrap"
          >
            <div
              class="mocked-styled-40 el-input-group"
            >
              <input
                aria-label="Input type email"
                class="mocked-styled-38 el-input"
                id="test-static-id"
                name="userEmail"
                type="email"
              />
              <label
                class="mocked-styled-41 el-label"
                for="test-static-id"
              >
                User Email
              </label>
            </div>
          </div>
          <div
            class="mocked-styled-133 el-input-wrap"
          >
            <div
              class="mocked-styled-40 el-input-group"
            >
              <label
                class="mocked-styled-41 el-label"
              >
                Subscription Type
              </label>
              <select
                aria-label="Select an item from the dropdown list"
                class="mocked-styled-83 el-select"
                name="subscriptionType"
              >
                <option
                  value=""
                >
                  Please Select
                </option>
                <option
                  value="applicationListing"
                >
                  Application Listing
                </option>
                <option
                  value="developerRegistration"
                >
                  Developer Registration
                </option>
                <option
                  value="developerEdition"
                >
                  Developer Edition
                </option>
              </select>
            </div>
          </div>
          <div
            class="mocked-styled-133 el-input-wrap"
          >
            <div
              class="mocked-styled-40 el-input-group"
            >
              <label
                class="mocked-styled-41 el-label"
              >
                Status
              </label>
              <select
                aria-label="Select an item from the dropdown list"
                class="mocked-styled-83 el-select"
                name="status"
              >
                <option
                  value=""
                >
                  Please Select
                </option>
                <option
                  value="active"
                >
                  Active
                </option>
                <option
                  value="cancelled"
                >
                  Cancelled
                </option>
              </select>
            </div>
          </div>
        </div>
      </form>
      <div
        class="el-mb11"
      >
        <div
          class="mocked-styled-2 el-button-group"
        >
          <div
            class="mocked-styled-3 el-button-group-inner"
          >
            <button
              class="el-intent-primary el-button"
              type="button"
            >
              <div
                class="mocked-styled-0 el-button-loader"
              />
              Download CSV
            </button>
          </div>
        </div>
      </div>
      <div
        class="mocked-styled-44 el-loader-container"
      >
        <div
          class="mocked-styled-46 el-loader"
        >
          <div
            class="mocked-styled-47 el-loader-moving-bar"
          />
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
